library/modules/Gui.cpp

```diff
@@ -934,12 +946,63 @@ df::unit *Gui::getAnyUnit(df::viewscreen *top)
         return NULL;
     }
 
+    if (VIRTUAL_CAST_VAR(screen, df::viewscreen_reportlistst, top))
+        return vector_get(screen->units, screen->cursor);
+
+    if (VIRTUAL_CAST_VAR(screen, df::viewscreen_announcelistst, top))
```

Comment:
This is also the main `a` screen, right? It looks like what you have handles `r`+Enter, which is good, but might not behave as expected on the `a` screen (disregard if you've tested that).

Reply:
I had not thought of that. But I could not find a way to fetch unit from a screen reports, so i left that case blank.

Reply:
Ok, looks good. There was some discussion about that in the DFHack thread and I don't believe there is a way that we know of.